### PR TITLE
Upgrade pyramid from 1.10 to 2.0

### PR DESCRIPTION
**Pull request by linrock**
Used these two references to figure out the new APIs. I didn't find any useful upgrade guides otherwise:
- [pyramid sessions reference](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/sessions.html)
- [pyramid.security API reference](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/security.html)

Fixes https://github.com/glinscott/fishtest/issues/887

### CODE REVIEW

**Comment by linrock** [comment]
just noticed the tests view pages crash with this error, so the new API for `has_permission` must be something other than `view_execution_permitted`:

image

**Comment by linrock** [comment]
The fixes required for the pyramid 1.10 to 2.0 upgrade appear to be:
- replace `UnencryptedCookieSessionFactoryConfig` with `SignedCookieSessionFactory`
- replace `authenticated_userid(request)` with `request.authenticated_userid`
- replace `has_permission("approve_run", request.context, request)` with `request.has_permission("approve_run")`

**Comment by ppigazzini** [comment]
DEV updated, start testing.

File: server/setup.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 CHANGES = ""
 
 requires = [
-    "pyramid==1.10",
+    "pyramid==2.0",
```
Comment (ppigazzini) [inline server/setup.py:7]: Better without version so I will able to catch with DEV a problem related to a new version. 
```python
requires = [
    "pyramid",
```

**Comment by ppigazzini** [comment]
DEV looks good:
 - registered a new user
 - admin is able to unblock/block another user
 - user is able to change email/password
 - user is able to create/stop/modify/delete his own test
 - user is not able to approve his own test
 - user is not able to stop/modify/delete the test of another user
 - user is not able to purge a test
 - admin is able to approve/stop/modify/purge test of another user
 - admin is not able to approve his own test
 - user is able to upload a net

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated, thank you @linrock :)

**Pull request merged**